src/mutation_input_structs.rs

```diff
@@ -0,0 +1,40 @@
+use async_graphql::{InputObject, SimpleObject};
+use bson::Uuid;
+use std::{
+    cmp::Ordering,
+    collections::{BTreeSet, HashSet},
+};
+
+#[derive(SimpleObject, InputObject)]
+pub struct CreateOrderInput {
+    /// UUID of user owning the order.
+    pub user_id: Uuid,
+    /// OrderItems of order.
+    pub order_items: BTreeSet<OrderItemInput>,
+}
+
+#[derive(SimpleObject, InputObject, PartialEq, Eq)]
+pub struct OrderItemInput {
+    /// UUID of product item associated with order item.
+    pub product_item_id: Uuid,
+    /// UUID of product variant version associated with order item.
+    pub product_variant_version_id: Uuid,
+    /// UUID of tax rate version associated with order item.
+    pub tax_rate_version_id: Uuid,
```

Comment:
imho this is not the job of the API user
you should save this for the product_variant_version in your db